lib/eet/client.rb

```diff
@@ -1,18 +1,45 @@
 module Eet
   class Client
-    attr_reader :certificate, :data
+  #Usage:
+    #client = Client.new(eet_certificate, prepared_eet_data)
+
+    #if you need pkp and bkp values before you actually register then you do need to run this first:
+
+    #client.prepare_message
+
+    #pkp and bkp values are exposed through message instance on the client:
+
+    #client.message.pkp
+    #client.message.bkp
+
+    #now you can continue with register action:
+
+    #client.register(:production)
+
+    attr_reader :certificate, :data, :message, :signed_message
 
     def initialize(certificate, data)
       @certificate = certificate
       @data = data
+      @message = nil
+      @signed_message = nil
     end
 
     def submit(environment)
-      message = Eet::Message.new(data)
-      message.pkp = Eet::Utils.create_pkp(message, certificate)
-      message.bkp = Eet::Utils.create_bkp(message.pkp)
+      prepare_message
+      register(environment)
+    end
+
+    def prepare_message
+      @message = Eet::Message.new(data)
+      @message.pkp = Eet::Utils.create_pkp(message, certificate)
+      @message.bkp = Eet::Utils.create_bkp(message.pkp)
+
+      @signed_message = Eet::Utils.sign(message.to_xml, certificate)
+    end
 
-      signed_message = Eet::Utils.sign(message.to_xml, certificate)
+    def register(environment)
+      return unless signed_message
```

Comment:
Nejsem si jistej timhle returnem. co by se teoreticky stalo, kdybyhom nemeli signed_message a zavolali #register? Tak se posle nejaka blbost a eet ji odmitne ne? To mi mozna prijde lepsi nez takovyhle silent fail?